Assets/Scripts/Managers/SaveManager.cs

```diff
@@ -0,0 +1,173 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using Buildings;
+using Entity.Player;
+using GameTime;
+using Interactables;
+using Inventory;
+using UnityEngine;
+using UnityEngine.AI;
+using Utils.Saves;
+
+namespace Managers
+{
+    public class SaveManager : Singleton<SaveManager>
+    {
+        /// <summary>
+        /// saves the current state of the game to JSON
+        /// </summary>
+        /// <param name="fileName"></param>
+        public void Save(string fileName)
+        {
+            Debug.Log("save initiated");
+            try
+            {
+                // grab required components to extract data from
+                GameObject player = PlayerManager.Instance.GetPlayer();
+                InventoryController inventoryController = InventoryManager.Instance.GetInvController();
+
+                // this is the player's current position in the map
+                Vector3 playerPosition = player.transform.localPosition;
+
+                // get health, hunger, thirst, sanity, ... from here
+                PlayerState state = player.GetComponent<PlayerState>();
+
+                //get a json-able list of items in the player's inventory
+                List<Item> inventoryData = inventoryController.GetItems();
+
+                //get all campsites
+                List<Campsite> campsites = CampsiteManager.Instance.campsites;
+
+                //get all buildings
+                Building[] buildings = FindObjectsOfType<Building>();
+                BuildingBlueprint[] blueprints = FindObjectsOfType<BuildingBlueprint>();
+
+                Clock clock = DayNightManager.Instance.GetComponent<Clock>();
+                float time = clock.TimeOfDay;
+                int daynumber = clock.Days;
+
+                //get all harvestables
+                Harvestable[] harvestables = FindObjectsOfType<Harvestable>();
+                List<SavedHarvestable> sh = new List<SavedHarvestable>();
+                foreach (Harvestable h in harvestables)
+                {
+                    sh.Add(new SavedHarvestable(h, h.IsRespawning, h.RespawnTimePassed));
+                }
+
+                // build a JSON-Object
+                Save save = new Save();
+
+                save.SetPlayerState(playerPosition, state.GetHealth(), state.GetSaturation(), state.GetHydration(),
+                    state.GetSanity());
+                save.SetWorldState(time, daynumber);
+                save.SetInventory(inventoryData);
+                save.buildVersion = Application.version;
+
+                save.campsites = new List<SavedCampsite>();
+                campsites.ForEach(cs => save.campsites.Add(new SavedCampsite(cs, cs.IsUnlocked)));
+
+                save.harvestables = sh;
+
+                Write(save, fileName);
+            }
+            catch (Exception e)
+            {
+                Debug.Log(e);
```

Comment:
Why is this a try catch?

Reply:
Writing to file might fail for unknown reasons (e.g. no write permissions)

Reply:
loading an incompatible save will result in a horrible amount of stuff that might go wrong, therefore try catch